nim_status/protocol/application_metadata_message.nim

```diff
@@ -0,0 +1,14 @@
+import # vendor libs
+  protobuf_serialization
+
+export protobuf_serialization
+
+# nim-status protobuf spec
+# automatically exports `type ApplicationMetadataMessage`
+import_proto3 "protobuf/application_metadata_message.proto"
+
+proc decode*(T: type ApplicationMetadataMessage, input: seq[byte]): T =
+  Protobuf.decode(input, T)
+
+proc encode*(input: ApplicationMetadataMessage): seq[byte] =
+  Protobuf.encode(input)
```

Comment:
Are these really necessary? Wouldn't this be sort of "ungenericizing" (or whatever the word is) the already generic `Protobuf.encode`/`Protobuf.decode`? It would mean that we would need to include these procs for each protobuf object introduced which can be more difficult to maintain.

Reply:
It allows for this:
```nim
let amm = ApplicationMetadataMessage()

let
  enc = amm.encode
  dec = ApplicationMetadataMessage.decode(enc)
```
Without it:
```nim
let amm = ApplicationMetadataMessage()

let
  enc = Protobuf.encode(amm)
  dec = Protobuf.decode(enc, ApplicationMetadataMessage)
```
I prefer the fist style. But:
> It would mean that we would need to include these procs for each protobuf object...

It's true we'd need them for each protobuf, but once we have them in a corresponding `.nim` module for each `.proto` spec, they are unlikely to change over time, even if we update the specs, i.e. copy over updated specs from status-go. So the effort to create them and re-export them from `nim_status/protocol.nim` seems like a mostly fixed development cost.